server/mscalendar/settings_daily_summary.go

```diff
@@ -0,0 +1,259 @@
+package mscalendar
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/store"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/settingspanel"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+type dailySummarySetting struct {
+	title       string
+	description string
+	id          string
+	dependsOn   string
+	optionsH    []string
+	optionsM    []string
+	optionsAPM  []string
+	store       settingspanel.SettingStore
+	getTimezone func(userID string) (string, error)
+}
+
+func NewDailySummarySetting(inStore settingspanel.SettingStore, getTimezone func(userID string) (string, error)) settingspanel.Setting {
+	os := &dailySummarySetting{
+		title:       "Daily Summary",
+		description: "When do you want to receive the daily summary?\n If you update this setting, it will automatically update to your the timezone currently set on your calendar.",
+		id:          store.DailySummarySettingID,
+		dependsOn:   "",
+		store:       inStore,
+		getTimezone: getTimezone,
+	}
+	os.optionsH = []string{}
+	for i := 0; i < 12; i++ {
+		os.optionsH = append(os.optionsH, fmt.Sprintf("%d", i))
+	}
+
+	os.optionsM = []string{}
+	for i := 0; i < 4; i++ {
+		os.optionsM = append(os.optionsM, fmt.Sprintf("%02d", i*15))
+	}
+
+	os.optionsAPM = []string{"AM", "PM"}
+
+	return os
+}
+
+func (s *dailySummarySetting) Set(userID string, value interface{}) error {
+	_, ok := value.(string)
+	if !ok {
+		return errors.New("trying to set Daily Summary Setting without a string value")
+	}
+	err := s.store.SetSetting(userID, s.id, value)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *dailySummarySetting) Get(userID string) (interface{}, error) {
+	value, err := s.store.GetSetting(userID, s.id)
+	if err != nil {
+		return nil, err
+	}
+
+	_, ok := value.(*store.DailySummaryUserSettings)
+	if !ok {
+		return nil, errors.New("current value is not a Daily Summary Setting")
+	}
+
+	return value, nil
+}
+
+func (s *dailySummarySetting) GetID() string {
+	return s.id
+}
+
+func (s *dailySummarySetting) GetTitle() string {
+	return s.title
+}
+
+func (s *dailySummarySetting) GetDescription() string {
+	return s.description
+}
+
+func (s *dailySummarySetting) GetDependency() string {
+	return s.dependsOn
+}
+
+func (s *dailySummarySetting) GetSlackAttachments(userID, settingHandler string, disabled bool) (*model.SlackAttachment, error) {
+	title := fmt.Sprintf("Setting: %s", s.title)
+	currentValueMessage := "Disabled"
+
+	actions := []*model.PostAction{}
+	if !disabled {
```

Comment:
This `!disabled` block is pretty long, leading up to just about the end of the function. Do you think we can instead have a short `if disabled` block that returns early on?